### PR TITLE
Get authentication token through API

### PR DESCRIPTION
**Pull request by Aratz**
**What problems does this PR solve?**
DDS imposes strong constraints on who can access the authentication token. For instance, as of now it is not possible to set group permissions on the token.

This PR makes it possible to send the token directly through the API call or to place it in a shared folder on Miarka.

**How has the changes been tested?**
 Unit tests have been adapted and more tests will be performed in staging.

**Reasons for careful code review**
If any of the boxes below are checked, extra careful code review should be inititated.

  - [ ] This PR contains code that could remove data

### CODE REVIEW

**Comment by Aratz** [comment]
I'll add a few more commits to allow both modes of action (with token file on Miarka and with token string through the API)

**Review by matrulda — COMMENTED** [review]
Looks good to me! :+1: Only one comment, let me know if you want to address it.

File: delivery/handlers/dds_handlers.py [hunk 1]
```diff
@@ -37,16 +40,30 @@ async def post(self, project_name):
...
         """
 
-        required_members = ["token_path"]
-        project_metadata = self.body_as_object(required_members=required_members)
+        required_members = ["auth_token"]
+        project_metadata = self.body_as_object(
+                required_members=required_members)
 
-        dds_project_id = await self.dds_service.create_dds_project(project_name, project_metadata)
+        with tempfile.NamedTemporaryFile(mode='w', delete=True) as token_file:
```
[the remark this answers is not in the document]
Comment (Aratz) [reply delivery/handlers/dds_handlers.py:53]: Oh, good point! Although I'm slightly more inclined to merge this now and do this over the summer (I can create a ticket for that). What do you think?

[the remark this answers is not in the document]
Comment (matrulda) [reply delivery/handlers/dds_handlers.py:53]: Sounds good :+1:

**Review by matrulda — APPROVED** [review]

**Pull request merged**